File: ds_8_matrix/L48-m.cpp
//
// Created by LiHeng on 2020/4/22.
//
#include <iostream>
#include <vector>
using namespace std;


//先转置；再前后对称互换列
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        for(int i=0 ; i<matrix.size(); ++i){
            for (int j=i+1; j<matrix[0].size(); ++j)
                swap(matrix[i][j],matrix[j][i]); //一开始犯了个巨大错误，c++中某个元素应该是matrix[i][j] ，不是matrix[i，j]
        }
        for(int i=0 ; i<matrix.size(); ++i){
            for (int j=0; j<matrix[0].size()/2; ++j)
                swap(matrix[i][j],matrix[i][matrix[0].size()-j-1]);
        }
    }
};

File: python-algorithm/ds_8_matrix/L48-m.py
# -*- coding: utf-8 -*-
# @Time    : 2020/10/4 10:06
# @Author  : No Name

# 先转置矩阵，再对称交换所有列
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        m, n = len(matrix), len(matrix[0])
        for i in range(m):
            for j in range(i, n):
                if j > i:
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
        for i in range(n // 2):
            for j in range(m):
                matrix[j][i], matrix[j][n - i - 1] = matrix[j][n - i - 1], matrix[j][i]
